osu.Game/Graphics/UserInterface/Counter.cs

```diff
@@ -0,0 +1,110 @@
+
+        public override string Name { get; }
+        public KeyCounter ParentCounter { get; set; }
+        public int Counts { get; private set; }
+
+        private bool isLit;
+        public bool IsLit
+        {
+            get { return isLit; }
+            protected set
+            {
+                if (isLit != value)
+                {
+                    isLit = value;
+                    UpdateGlowSprite();
+                    if (value && ParentCounter.IsCounting)
+                        IncreaseCount();
+                }
+            }
+        }
+
+        public Color4 KeyDownTextColor { get; set; } = Color4.DarkGray;
+        public Color4 KeyUpTextColor { get; set; } = Color4.White;
+
+        protected Count(string name)
+        {
+            Name = name;
+        }
+
+        public override void Load()
+        {
+            base.Load();
+            Children = new Drawable[]
+            {
+                buttonSprite = new Sprite
+                {
+                    Texture = Game.Textures.Get(@"KeyCounter/key-up"),
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    Children = new Drawable[]
+                    {
+                        glowSprite = new Sprite
+                        {
+                            Texture = Game.Textures.Get(@"KeyCounter/key-glow"),
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                        }
+                    }
+                },
+                keySpriteText = new SpriteText
+                {
+                    Text = Name,
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    Position = new Vector2(0, -buttonSprite.Height / 4),
+                    Colour = KeyUpTextColor
+                },
+                countSpriteText = new SpriteText
+                {
+                    Text = Counts.ToString(),
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    Position = new Vector2(0, buttonSprite.Height / 4),
+                    Colour = KeyUpTextColor
+                }
+            };
+            glowSprite.Hide();
+        }
+
+        private void UpdateGlowSprite()
+        {
+            //can have a FadeTime property or const
+            if (IsLit)
+            {
+                glowSprite.Show();
+                countSpriteText.FadeColour(KeyDownTextColor, 0);
+                keySpriteText.FadeColour(KeyDownTextColor, 0);
+            }
+            else
+            {
+                glowSprite.Hide();
+                countSpriteText.FadeColour(KeyUpTextColor, 0);
+                keySpriteText.FadeColour(KeyUpTextColor, 0);
+            }
+        }
+
+        private void IncreaseCount()
+        {
+            Counts++;
+            countSpriteText.Text = Counts.ToString();

```

Comment:
Probably wanna consider formatting with separator characters for numbers >= 1000 (`1,000`).